packages/imitation_game/tests/smiley/flutter/smiley/lib/main.dart

```diff
@@ -0,0 +1,104 @@
+import 'dart:convert' show jsonEncode, utf8;
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:http/http.dart' as http;
+import 'package:poll_ios_stats/poll_ios_stats.dart';
+
+String _hostIp;
+
+Future<String> _getHostIp() async {
+  return await rootBundle.loadString('assets/ip.txt');
+}
+
+Future<void> _sendResult(double result) async {
+  assert(_hostIp != null);
+  print('sending result $result...');
+  final http.Response response = await http.post(
+    'http://$_hostIp',
+    headers: <String, String>{
+      'Content-Type': 'application/json; charset=UTF-8',
+    },
+    body: jsonEncode(<String, dynamic>{
+      'test': 'smiley',
+      'platform': 'flutter',
+      'results': <String, double>{
+        'startupTime': result
+      },
+    }),
+  );
+  if (response.statusCode != 200) {
+    print('error when posting results:${response.statusCode}');
+  }
+}
+
+void main() {
+  runApp(MyApp());
+  SystemChrome.setEnabledSystemUIOverlays([]);
+  _getHostIp().then((String ip) async {
+    _hostIp = ip;
+  });
+}
+
+class MyApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Flutter Demo',
+      theme: ThemeData(
+        primarySwatch: Colors.blue,
+        visualDensity: VisualDensity.adaptivePlatformDensity,
+      ),
+      home: MyHomePage(),
+    );
+  }
+}
+
+class MyHomePage extends StatefulWidget {
+  MyHomePage({Key key}) : super(key: key);
+
+  @override
+  _MyHomePageState createState() => _MyHomePageState();
+}
+
+class _MyHomePageState extends State<MyHomePage> {
+  final AssetImage _image = const AssetImage('images/smiley.png');
+  bool _loading = true;
+
+  @override
+  void initState() {
+    _image
+        .resolve(const ImageConfiguration())
+        .addListener(ImageStreamListener((_, __) {
+      if (mounted) {
+        setState(() {
+          _loading = false;
+          WidgetsBinding.instance.addPostFrameCallback((_) async {
```

Comment:
ditto, add code comment for what this is for?

Reply:
done